crates/wasmtime/src/config.rs

```diff
@@ -1093,6 +1095,19 @@ impl Config {
         self
     }
 
+    /// Configure whether deserialized modules should validate version
+    /// information. This only effects [`Module::from_bytes`], which is used to
+    /// load compiled code from trusted sources.  When true,
+    /// [`Module::from_bytes`] verifies that the wasmtime crate's
+    /// `CARGO_PKG_VERSION` matches with the version in the binary, which was
+    /// produced by [`Module::serialize`] or [`Engine::precompile_module`].
+    ///
+    /// This value defaults to true.
+    pub fn validate_module_version(&mut self, check: bool) -> &mut Self {
```

Comment:
To bikeshed this slightly, this sounds like it's dealing with the wasm binary's version header in one way or another so I think we want to scope this differently perhaps. Some other possibilities:

* `precompiled_module_emit_version`
* `deserialize_check_wasmtime_verison`
* `omit_wasmtime_version`

or... others?

Reply:
I didn't mess with the serialization side because I figured it only changed behavior at deserialization, so it wasn't harmful to leave a version string in the binary that we didn't care about. But it is easy enough to turn it off emitting it if you think it is useful (maybe for someone who wants to binary diff modules across different builds?)

I like `deserialize_check_wasmtime_version` as a name for what is implemented - module has a lot of meanings so that is more clear.

Reply:
renamed c27e152

Reply:
Nah yeah this seems fine to have for now. Can you add a test for this as well?